trade-finance-manager-ui/scripts/moj-filter-hide.ts

```diff
@@ -0,0 +1,52 @@
+import $ from 'jquery';
+
+declare global {
+  interface Window {
+    MOJFrontend: any;
+  }
+}
+
+type FilterVisibility = 'SHOWN' | 'HIDDEN';
+
+const filterSelector = '.moj-filter-layout__filter';
+const toggleButtonContainerSelector = '.moj-action-bar__filter';
+const toggleButtonContainer = document.querySelector(toggleButtonContainerSelector)!;
+
+const params = (toggleButtonContainer as HTMLElement).dataset;
+const FILTER_VISIBILITY_SESSION_KEY = `${params.filteredTableId}-filters-visibility-state`;
+
+const setFilterVisibility = (visibility: FilterVisibility) => {
+  sessionStorage.setItem(FILTER_VISIBILITY_SESSION_KEY, visibility);
+};
+
+const cachedShownStatus = sessionStorage.getItem(FILTER_VISIBILITY_SESSION_KEY) as FilterVisibility | null;
+
+if (cachedShownStatus) {
+  setFilterVisibility(cachedShownStatus);
+}
+
+// Initially display filters, but hide them if they were previously hidden
+const startHidden = cachedShownStatus === 'HIDDEN';
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-call, no-new, @typescript-eslint/no-unsafe-member-access
+new window.MOJFrontend.FilterToggleButton({
```

Comment:
The new keyword is necessary here because it creates a new instance of the FilterToggleButton class, which sets up event listeners and modifies the DOM to enable the filter toggle behavior. Disabled in line with `portal/scripts/moj-filters.js`.